test/unit_tests/braket/experimental/autoqasm/test_api.py

```diff
@@ -839,3 +839,13 @@ def ghz(n: int):
 cnot __qubits__[0], __qubits__[4];"""
 
     assert make_ghz(5).to_ir() == expected
+
+
+def test_double_decorated_function():
+    @aq.function
+    @aq.function
+    def empty_program() -> None:
+        pass
```

Comment:
nice! I mean this is obviously a silly example, but in cases where the decorator is applied programmatically, ensuring this works could help to avoid paper cuts.